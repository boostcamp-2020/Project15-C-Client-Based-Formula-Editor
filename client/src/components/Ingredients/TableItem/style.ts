import styled from '@emotion/styled';
interface FormulaItemStyleProps {
  image?: string;
  isSvg: boolean;
}

export const FormulaItem = styled.li<FormulaItemStyleProps>`
  background: url(${(props) => `./image/${props.image}`});
  background-size: 100%;
  width: ${(props) => (!props.isSvg ? '35px' : props.image?.includes('b') ? '140px' : '70px')};
  height: ${(props) => (!props.isSvg ? '35px' : '70px')};
  font-size: 1rem;
  text-align: center;
  line-height: 35px;
  background-position: center;
  cursor: pointer;
  border: none;
  border-right: 0.5px solid #d5d5d5;
  border-bottom: 0.5px solid #d5d5d5;
  background-repeat: no-repeat;
  transition: font-size 0.3s, color 0.3s, background-size 0.3s;

  &:hover {
    background-size: 120%;
    border: 0.5px solid black;
    filter: invert(63%) sepia(13%) saturate(2335%) hue-rotate(87deg) brightness(91%) contrast(83%);
    font-size: 1.5rem;
  }
`;
