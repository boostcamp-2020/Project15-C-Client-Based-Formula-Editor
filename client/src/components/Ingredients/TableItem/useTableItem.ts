import { RootState } from '@contexts/index';
import { useSelector } from 'react-redux';

const useTableAreaItem = () => {
  const { mathfield } = useSelector((state: RootState) => state.latex);

  const onChangeHandler = (latex: string) => {
    mathfield?.write(latex);
  };

  return { onChangeHandler };
};

export default useTableAreaItem;
