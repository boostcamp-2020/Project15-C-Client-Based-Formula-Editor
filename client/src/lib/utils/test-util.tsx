import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, Store } from 'redux';
import { Provider } from 'react-redux';
import reducer, { RootState } from '../../contexts';

function render(
  ui: React.ReactElement,
  {
    initialState = {
      latex: {
        maxId: 0,
        currentTab: 0,
        totalLatex: [
          {
            id: 0,
            latex: '',
            fontSize: '15',
            fontColor: '#000000',
            textAlign: 'left',
          },
        ],
        mathfield: null,
        mathfieldRef: null,
      },
      user: {
        loading: false,
        error: null,
        userInfo: {
          userId: null,
          favoriteLists: [],
        },
      },
    },
    store = createStore(reducer, initialState),
  }: { initialState?: RootState; store?: Store } = { initialState: undefined, store: undefined }
) {
  function Wrapper({ children }: { children?: React.ReactNode }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
