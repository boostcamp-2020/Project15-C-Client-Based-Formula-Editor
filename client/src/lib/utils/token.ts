const USER_TOKEN = 'USER_TOKEN';

export const setToken = (userToken: string): void => {
  chrome.storage.sync.set({ USER_TOKEN: userToken }, function () {});
};

export const getToken = () => {
  return new Promise((resolve, reject) => {
    chrome.storage.sync.get([USER_TOKEN], function (result) {
      resolve(result[USER_TOKEN]);
    });
  });
};
