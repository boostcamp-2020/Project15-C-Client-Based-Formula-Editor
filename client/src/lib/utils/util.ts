import { Action } from 'redux';

export function createAction<T, P>(type: T): (payload: P) => Action<T> & { payload: P };
export function createAction<T>(type: T): () => Action<T>;
export function createAction<T, P>(type: T) {
  return (payload?: P) => {
    return {
      type,
      payload,
    };
  };
}
