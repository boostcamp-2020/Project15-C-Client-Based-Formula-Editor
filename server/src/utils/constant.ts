export enum STATUS_CODE {
  SUCCESS = 200,
  CREATED = 201,
  CLIENT_ERROR = 400,
  UNAUTHORIZED = 401,
  SERVER_ERROR = 500,
}

export enum ERROR_MESSAGE {
  CLIENT_ERROR = 'Request Error',
  UNAUTHORIZED_ERROR = 'Unauthorized Error',
  SERVER_ERROR = 'Server Error',
}
