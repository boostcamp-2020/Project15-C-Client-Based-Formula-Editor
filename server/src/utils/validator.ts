export enum ValidateType {
  Number = 'number',
  String = 'string',
  Object = 'object',
  Array = 'array',
  Boolean = 'boolean',
  Undefined = 'undefined',
  Null = 'null',
}

export function isValidateType<T>(
  target: T,
  type: ValidateType,
  callback?: (target: T) => boolean
): boolean {
  if (type === ValidateType.Null && target !== null) {
    return false;
  }

  if (type === ValidateType.Number && !Number.isFinite(Number(target))) {
    return false;
  }

  if (type === ValidateType.Array && !Array.isArray(target)) return false;

  if (
    (type === ValidateType.Object ||
      type === ValidateType.String ||
      type === ValidateType.Boolean ||
      type === ValidateType.Undefined) &&
    typeof target !== type
  )
    return false;

  if (callback && !callback(target)) return false;

  return true;
}
