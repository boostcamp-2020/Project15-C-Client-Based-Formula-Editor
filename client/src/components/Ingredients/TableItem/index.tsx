import React from 'react';
import { Header, Table, Label } from 'semantic-ui-react';
import useTextAreaItem from './useTableItem';
import * as S from './style';
import { getFavorites } from '../../../lib/apis/favorite';

interface DataType {
  id: number;
  title: string;
  latex: string;
}

interface TableItemType {
  headerTitle: string;
  headerLatex: string;
  data: DataType[];
}

const removeHandler = async (id: number) => {
  console.log('지우기');
};

function TableItem({ headerTitle, headerLatex, data }: TableItemType) {
  const { onChangeHandler } = useTextAreaItem();
  const dataList = data.map((item, index) => (
    <Table.Row key={index}>
      <Table.Cell width={1}>
        <Header.Content>{item.title}</Header.Content>
      </Table.Cell>
      <Table.Cell width={4}>
        <S.TableContainer>
          <S.TableSpan onClick={() => onChangeHandler(item.latex)}>{item.latex}</S.TableSpan>
          <S.TableWrapper>
            <Label size="mini" color="teal" onClick={() => removeHandler(item.id)}>
              X
            </Label>
          </S.TableWrapper>
        </S.TableContainer>
      </Table.Cell>
    </Table.Row>
  ));

  return (
    <Table basic="very" celled collapsing fixed unstackable>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell>{headerTitle}</Table.HeaderCell>
          <Table.HeaderCell>{headerLatex}</Table.HeaderCell>
        </Table.Row>
      </Table.Header>
      <Table.Body>{dataList}</Table.Body>
    </Table>
  );
}

export default TableItem;
