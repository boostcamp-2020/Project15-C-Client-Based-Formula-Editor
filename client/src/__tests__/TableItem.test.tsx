import * as React from 'react';
import { render } from '../lib/utils/test-util';
import TableItem from '../components/Ingredients/TableItem/index';
import '@testing-library/jest-dom';

describe('<TableItem />', () => {
  const data = [{ id: 1, latex: '1+1', title: '나의 latex' }];
  it('renders TableItem', () => {
    const { container } = render(
      <TableItem headerTitle={'title'} headerLatex={'latex'} data={data} />
    );
    expect(container).toMatchSnapshot();
  });
  it('in Table', () => {
    const { container, getByText } = render(
      <TableItem headerTitle={'title'} headerLatex={'latex'} data={data} />
    );
    expect(container.querySelector('table')).toBeInTheDocument;
    const header = getByText('title');
    expect(header).toBeInTheDocument;
    const content = getByText('latex');
    expect(content).toBeInTheDocument;
  });
});
