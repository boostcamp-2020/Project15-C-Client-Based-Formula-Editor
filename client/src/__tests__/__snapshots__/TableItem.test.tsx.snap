// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableItem /> renders TableItem 1`] = `
<div>
  <table
    class="ui celled collapsing fixed unstackable very basic table"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class=""
        >
          title
        </th>
        <th
          class=""
        >
          latex
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="one wide"
        >
          <div
            class="content"
          >
            나의 latex
          </div>
        </td>
        <td
          class="four wide"
        >
          <div
            class="css-1ip1yi5"
          >
            <span
              class="css-3phne6"
            >
              1+1
            </span>
            <div
              class="css-n7fkea"
            >
              <div
                class="ui mini label"
              >
                X
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
